definition.go

```diff
@@ -1180,6 +1198,8 @@ func (gt *InputObject) defineFieldMap() InputObjectFieldMap {
 	return resultFieldMap
 }
 func (gt *InputObject) Fields() InputObjectFieldMap {
+	gt.mu.Lock()
```

Comment:
This functions I think needs to match the one for *Object. That is, if gt.fields is nil then it needs to be rebuilt as AddInputField nils it out.

https://github.com/sprucehealth/graphql/blob/c9b790deef52145485f35e397e7f4c18f49cb75d/definition.go#L445

Reply:
Good call. I think input object handles this in (*InputObject).defineFieldMap and just needs to be recalled if it's nil after an update. Made that change.